lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -908,19 +920,20 @@ def configure_network(self, vm_obj):
                 self.change_detected = True
 
             if vm_obj is None or self.should_deploy_from_template():
-                guest_map = vim.vm.customization.AdapterMapping()
-                guest_map.adapter = vim.vm.customization.IPSettings()
-                guest_map.adapter.ip = vim.vm.customization.FixedIp()
-                guest_map.adapter.ip.ipAddress = str(network_devices[key]['ip'])
-                guest_map.adapter.subnetMask = str(network_devices[key]['subnet_mask'])
+                if 'ip' in self.params['networks'][network]:
```

Comment:
I just tested this use case now and it seems to be ok. I can create a VM from scratch with this network parameters:
```
    networks:
      my_network_interface:
        vlan: 163
```